compose/service.py

```diff
@@ -477,9 +486,6 @@ def _next_container_number(self, one_off=False):
         return 1 if not numbers else max(numbers) + 1
 
     def _get_links(self, link_to_self):
-        if self.use_networking:
-            return []
-
```

Comment:
I think we still need a check here, otherwise `run` is going to create the `link_to_self`.  It can probably just be `is self.networks: return []` ?